### ISSUE TITLE
Continue/ resume training: update learning rate

### ISSUE THREAD

**Issue by Bowen-Lu** [junyanz/pytorch-CycleGAN-and-pix2pix#1095]
Hello, could you please tell me how to set the right learning rate for the first epoch in continue training? 
When I want from 151 epoch to go on training (100 epochs for lr_decay), the model wird reset the lr to initial value with 0.0002. That means for the epoch 151, lr = 0.0002 wird used. 
If I change --lr in options to 0.0001, does it affect further learning rate?  (lr = lr_lambda * initial_lr)
Will the initial_lr be 0.0001?
Any reply will be appreciated.

**Comment by junyanz** [comment]
You could use a new `--lr`. It should affect it.

**Comment by Bowen-Lu** [comment]
But I want the initial learning rate to keep the same. In an other word, lr = lr_lambda * 0.0002 for epoch 151~200, not the lr = lr_lambda * 0.0002. Does it work, that we simply change the --lr?
lr_lambda = 1.0 - max(0, epoch + epoch_count - .n_epochs) / n_epochs_decay + 1);

**Comment by junyanz** [comment]
Got your question. If you set `--epoch_count 150`, you can keep the original learning rate.

**Comment by Bowen-Lu** [comment]
If we set --epoch_count 150, it will be training from epoch 150. But for epoch 150, the learning rate is still initial value 0.0002, expected is 0.0002 * (1 - (150 - 100) / 101). Because model.update_learning_rate() musst be implemented after model.optimize_parameters(). I tried update learning rate before training, it raises error in PyTorch. That is to say, the first epoch of continue-training can not use the right value of learning rate. 

- first train: --n_epoch 100 --n_epochs_decay 100 --lr 0.0002 --epoch_count 1 
- continue: --n_epoch 100 --n_epochs_decay 100 --lr 0.0002 --epoch_count 150 --epoch 149 --continue_train

**Comment by junyanz** [comment]
I submitted a new [commit](https://github.com/junyanz/pytorch-CycleGAN-and-pix2pix/commit/fd29199c33bd95704690aaa16f238a4f8e74762c). The code will update the learning rate in the beginning of each epoch.

**Comment by Bowen-Lu** [comment]
When I get the old_lr i.e. current learning rate before continue training, I found it's right value, not the initial value 0.0002 as I thought. You pervious code is exactly right. Sorry for that and appreciated for your help. My issue was perhaps because of the overlapping of loss record. (I used tensorboard to record loss behaviour in colab).

**Comment by arrrrr3186** [comment]
let's say, I want to continue the training process from 151th epoch, and  i want my model to read the 150th epoch's LR, is there a way to automate the step or we have to do it externally. 

Like this: 
    first train: --n_epoch 100 --n_epochs_decay 100 --lr 0.0002 --epoch_count 1
    continue: --n_epoch 100 --n_epochs_decay 100 --lr 0.0001253 --epoch_count 150 --epoch 149 --continue_train
